src/page-multi-facility/ReSyncRefFacilityModal.tsx

```diff
@@ -0,0 +1,266 @@
+import { StyledButton } from "../design-system/InputButton";
+import InputSelect from "../design-system/InputSelect";
+import ModalDialog from "../design-system/ModalDialog";
+import { useFacilities } from "../facilities-context";
+import useRejectionToast from "../hooks/useRejectionToast";
+import useScenario from "../scenario-context/useScenario";
+import {
+  Facilities,
+  Facility,
+  FacilityReferenceMapping,
+  ModelInputs,
+  ReferenceFacility,
+} from "./types";
+
+const ModalContent = styled.div`
+  display: flex;
+  flex-direction: column;
+  font-weight: normal;
+`;
+
+const SaveButton = styled(StyledButton)`
+  font-size: 14px;
+  font-weight: normal;
+`;
+
+const CancelButton = styled(StyledButton)`
+  background: transparent;
+  color: ${Colors.forest};
+  font-family: "Libre Franklin";
+  font-size: 14px;
+  line-height: 26px;
+`;
+
+const ModalFooter = styled.div`
+  display: flex;
+  flex-flow: row nowrap;
+  justify-content: space-between;
+  margin-top: 20px;
+`;
+
+const SyncDataTitleContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const TitleText = styled.h1`
+  font-weight: 500;
+  font-size: 15px;
+  line-height: 22px;
+  color: ${Colors.forest};
+  font-family: "Libre Franklin";
+
+  &:nth-of-type(3) {
+    margin: 10px 0;
+  }
+`;
+
+const Row = styled.div`
+  border-bottom: 1px solid ${Colors.paleJade20};
+  color: ${Colors.forest}
+  cursor: pointer;
+  display: flex;
+  flex-flow: row nowrap;
+  font-family: "Libre Franklin", sans serif;
+  font-size: 15px;
+  line-height: 16px;
+  letter-spacing: -0.01em;
+  padding: 24px 0;
+`;
+
+const Header = styled.h3`
+  color: ${Colors.opacityForest};
+  flex: 1 1;
+  font-family: "Poppins", sans serif;
+  font-size: 9px;
+  font-weight: 600;
+`;
+
+const HeaderRow = styled(Row)`
+  padding: 10px 0;
+  border-bottom: 0;
+`;
+
+const FacilityName = styled.div`
+  flex: 1 1;
+`;
+
+const FacilitySelectContainer = styled.div`
+  max-width: 300px;
+  flex: 1 1;
+`;
+
+const Spacer = styled.span`
+  margin-right: 2em;
+`;
+
+const SyncDataTitle: React.FC<{
+  stateName: ModalProps["stateName"];
+  systemType: ModalProps["systemType"];
+}> = ({ stateName, systemType }) => (
+  <SyncDataTitleContainer>
+    <TitleText>Prepopulate Data</TitleText>
+    <TitleText>
+      We've found new facility data - select a corresponding facility to
+      autofill with case data
+    </TitleText>
+    {stateName && systemType && (
+      <TitleText>
+        State: {stateName}
+        <Spacer />
+        Type of System: {systemType}
+      </TitleText>
+    )}
+  </SyncDataTitleContainer>
+);
+
+const FacilitiesSelect: React.FC<{
+  value: Facility["id"] | undefined;
+  selections: { [key in ReferenceFacility["id"]]: Facility["id"] };
+  facilities: Facilities;
+  onChange: (value: Facility["id"] | undefined) => void;
+}> = ({ selections, value, facilities, onChange }) => {
+  const disabledOption = (facilityId: Facility["id"]) =>
```

Comment:
Good idea to disable the option for facility references that have already been selected 👍